meson.build

```diff
@@ -442,7 +442,8 @@ foreach ident : ['secure_getenv', '__secure_getenv']
 endforeach
 
 foreach ident : [
-        ['memfd_create',      '''#include <sys/memfd.h>'''],
+        ['memfd_create',      '''#define _GNU_SOURCE
+                                 #include <sys/mman.h>'''],
```

Comment:
Hmm, memfd_create(2) disagrees with glibc apparently. But if is actually in sys/mman.h, that's what we should check. If it turns up in other places in the future, we can just add more header files to the check.